unittests/allplatformstests.py

```diff
@@ -4584,13 +4584,25 @@ def test_symlinked_subproject(self):
 
     def test_configure_same_noop(self):
         testdir = os.path.join(self.unit_test_dir, '108 configure same noop')
-        self.init(testdir, extra_args=['-Dopt=val'])
+        args = [
+            '-Dstring=val',
+            '-Dboolean=true',
+            '-Dcombo=two',
+            '-Dinteger=7',
+            '-Darray=[\'three\']',
+            '-Dfeature=disabled'
+        ]
+        self.init(testdir, extra_args=args)
 
         filename = os.path.join(self.privatedir, 'coredata.dat')
         oldmtime = os.path.getmtime(filename)
-        self.setconf(["-Dopt=val"])
+        self.setconf(args)
         newmtime = os.path.getmtime(filename)
         self.assertEqual(oldmtime, newmtime)
+        oldmtime = newmtime
+        self.setconf(['-Dstring=abc'])
+        newmtime = os.path.getmtime(filename)
+        self.assertGreater(newmtime, oldmtime)
```

Comment:
I would also add a check for some built-in options such as prefix and buildtype which are special in this case.

Reply:
I would also verify that changing their value does change coredata file.

Reply:
I don't think you added that test yet, did you?